### ISSUE TITLE
metamorphic: TestMeta failed

### ISSUE THREAD

**Issue by nicktrav** [cockroachdb/pebble#2084]
https://teamcity.cockroachdb.com/viewLog.html?buildId=7247629&buildTypeId=Cockroach_Nightlies_Pebble_Metamorphic

```console
[TestMeta/random-024] === RUN   TestMeta/random-024

===== SEED =====
1667275606895981803
===== DIFF =====
/artifacts/meta/221101-040646.895/{standard-000,standard-001}
@@ -776,11 +776,11 @@
 iter3.Next("") // [false] <nil> #775
 iter4.Prev("lomcsomc") // [invalid] pebble: unsupported reverse prefix iteration #776
 iter3.Last() // [false] <nil> #777
 iter4.SeekPrefixGE("jpvd@4") // [true,"jpvd@4",<no point>,["jpvd","jpvd\x00")=>{"@5"="dtrllgyrscuehwdghes"}*] <nil> #778
 db.Set("lomcsomc", "lapovcwyxbfpuaaq") // <nil> #779
-iter4.SeekPrefixGE("wytbhdysdzr@8") // [false] <nil> #780
+iter4.SeekPrefixGE("wytbhdysdzr@8") // [true,"wytbhdysdzr@3","weynicucocyolwfgbd",<no range>*] <nil> #780
 db.Get("bwnftqyof@7") // ["ob"] <nil> #781
 iter5.SetBounds("", "hlwncuuzwde@5") // <nil> #782
 iter1.SeekPrefixGE("lomcsomc@5") // [true,"lomcsomc@5",<no point>,["lomcsomc","lomcsomc\x00")=>{""="jlkvwmyzdtgxmntfyke","@7"="tvxhonong"}] <nil> #783
 db.RangeKeyUnset("qkwbksxs", "xixntobf", "@10") // <nil> #784
 db.Merge("pizikizgly@6", "ftlbvmc") // <nil> #785
```

Repro with

```bash
$ go test -tags 'invariants' -test.v -run TestMeta$ ./internal/metamorphic -seed 1667275606895981803 -ops "uniform:5000-10000"
```

**Comment by jbowens** [comment]
Looks to be caused by the just-merged #2058. Will investigate today.

**Comment by nicktrav** [comment]
Confirmed that it looks like #2058:

```bash
$ git checkout master
$ git checkout HEAD~1
$ git rev-parse --short HEAD
b5deee5d
$ go test -tags 'invariants' -test.v -run TestMeta$ ./internal/metamorphic -seed 1667275606895981803 -ops "uniform:5000-10000"
...
PASS
ok      github.com/cockroachdb/pebble/internal/metamorphic      24.765s
```
